lib/tasks/deploy.rake

```diff
@@ -134,12 +134,19 @@ namespace :deploy do
   end
 
   task :rolling_deploy do
+    stop_start_errors = []
     on_each_docker_host do |server|
       service = defined_service
 
-      stop_containers(server, service, fetch(:stop_timeout, 30))
-
-      container = start_new_container(server, service, defined_restart_policy)
+      begin
+        stop_containers(server, service, fetch(:stop_timeout, 30))
+        container = start_new_container(server, service, defined_restart_policy)
+      rescue e
+        on_fail = fetch(:rolling_deploy_on_docker_failure, :exit)
+        raise e unless on_fail == :continue
```

Comment:
I think here you want just `raise` which will re-raise the last error. Otherwise you get the stacktrace from here rather than the original error.